File: app/src/main/kotlin/ru/tutu/stations/ui/mvp/presenter/MvpPresenter.java
package ru.tutu.stations.ui.mvp.presenter;

import android.support.annotation.NonNull;
import ru.tutu.stations.ui.mvp.view.MvpView;

/**
 * MVP презентер.
 *
 * @param <V> {@link MvpView}, c которой работает презентер.
 * @author Grigoriy Pryamov
 */
public interface MvpPresenter<V extends MvpView> {

    /**
     * Привязывает {@link MvpView} к презентеру
     *
     * @param view Представление
     */
    void bind(@NonNull V view);

    /**
     * Отвязывает {@link MvpView} от презентера
     *
     * @param view
     */
    void unbind(@NonNull V view);

    /**
     * Вызывается при очищении ссылки на презентер.
     * Последняя точка, в которой нужно отвязаться от всех static ссылок.
     */
    void destroy();
}

File: app/src/main/kotlin/ru/tutu/stations/ui/mvp/presenter/MvpPresenter.kt
package ru.tutu.stations.ui.mvp.presenter

import ru.tutu.stations.ui.mvp.MvpView

/**
 * MVP презентер.
 *
 * @param <V> {@link MvpView}, c которой работает презентер.
 * @author Grigoriy Pryamov.
 */
interface MvpPresenter<V : MvpView> {
    /**
     * Привязывает [MvpView] к презентеру
     *
     * @param view Представление
     */
    fun bind(view: V)

    /**
     * Отвязывает [MvpView] от презентера
     *
     * @param view
     */
    fun unbind(view: V)

    /**
     * Вызывается при очищении ссылки на презентер.
     * Последняя точка, в которой нужно отвязаться от всех static ссылок.
     */
    fun destroy()
}